azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -81,8 +83,23 @@ public void run() throws Exception {
     info(commands.size() + " commands to execute.");
     File[] propFiles = initPropsFiles();
     Map<String, String> envVars = getEnvironmentVariables();
-
+  
+    String nativeLibFolder = null;
+    String executeAsUserBinary = null;
+    String userToProxy = null;
+    boolean isExecuteAsUser = sysProps.getBoolean(EXECUTE_AS_USER, false);
+    
+    if(isExecuteAsUser){
```

Comment:
I've confirmed the flag to work, which will enable the worst case scenario of rolling back.
However, how to warn users beforehand and the impact analysis will be something harder to do